routes/endpoints/importRoutes.js

```diff
@@ -1,37 +1,183 @@
+                    line: thisLine,
+                };
+            if (!thisLine.owner)
+                throw {
+                    error: "Project requires an owner",
+                    line: thisLine,
+                };
+            if (!thisLine.contactInfo)
+                throw {
+                    error: "Project requires contact info",
+                    line: thisLine,
+                };
+
+            /**
+             * Parse tags into an array. Leave empty if unset
+             */
+            var tagArray = [];
+            if (thisLine.tags) {
+                tagArray = thisLine.tags.split(",");
+            }
+
+            /**
+             * Create default statuses and set with data from csv
+             */
+            var tempStatuses = {
+                isApproved: true,
+                isRecruiting: thisLine.recruiting,
+                isProposal: false,
+                isActive: false,
+                isPaused: false,
+                isStopped: false,
+                isArchived: false,
+            };
+            tempStatuses[thisLine.status] = true;
+
+            /**
+             * Fill an object with data matching db schema
+             */
+            var tempProject = {
+                name: thisLine.name,
+                owner: thisLine.owner,
+                statuses: tempStatuses,
+                contactInfo: thisLine.contactInfo,
+                description: thisLine.description,
+                gitRepo: thisLine.github,
+                tags: tagArray,
+                date: new Date(),
+            };
+
+            /**
+             * Optionally try to find an existing user by their EUID
+             */
+            var localUser;
+            if (thisLine.euid.match(euidReg)) {
+                localUser = await userModel.findOne({ euid: thisLine.euid });
+            }
+
+            /**
+             * If we didnt get a user from the euid, try the email
+             */
+            if (!localUser && thisLine.contactInfo.slice(thisLine.contactInfo.length - 7) == "unt.edu") {
+                localUser = await userModel.findOne({ emails: thisLine.contactInfo });
+            }
+
+            if (localUser) {
+                tempProject.ownerID = localUser._id;
+                tempProject.owner = localUser.name + " " + localUser.lastname;
+                tempProject.contactInfo = localUser.email;
+            }
+            var newProject = new projectModel(tempProject);
+            //commented not to save for real
+            //await newProject.save()
+            numSaved++;
+        } catch (err) {
+            if (err.line) {
+                errorMessages.push(err);
+            }
+        }
+    }
+
+    console.log({
+        savedFiles: numSaved,
+        errors: errorMessages,
+    });
+}
+
+createDocuments();
+router.post("/import", auth.admin, function (req, res) {
+    //get file from database later im tired
+});
+
+module.exports = router;
+
+// csvtojson()
```

Comment:
Is all of Jasmine's code deleted???

Reply:
Yes with her permission on a voice call because the structure needed to be different for error checking and parsing